src/serializers/type_serializers/function.rs

```diff
@@ -55,9 +57,13 @@ impl FunctionPlainSerializer {
         extra: &Extra,
     ) -> PyResult<PyObject> {
         let py = value.py();
+        let info = SerializationInfo::new(py, include, exclude, extra);
         if self.when_used.should_use(value, extra) {
-            self.func
-                .call1(py, (value, SerializationInfo::new(py, include, exclude, extra)))
+            if self.needs_model_instance {
+                self.func.call1(py, (extra.model.unwrap(), value, info))
```

Comment:
Unless I'm missing something, I think it would be better to actually check that extra.model is not None; don't want a schema generation bug to cause a rust panic.

I'm not quite sure yet how the needs_model_instance and extra.model values get plumbed, but I almost think you could change `if self.needs_model_instance` to `if let Some(model) = extra.model`; if you do find a way to produce a scenario where these conditions aren't equivalent, I would think you might even get a useful python error (saying something to the effect of 'you didn't provide enough arguments', or 'you provided too many arguments').